credential/manifest/model_test.go

```diff
@@ -215,61 +221,62 @@ func TestIsValidPair(t *testing.T) {
 			LDP: &exchange.LDPType{ProofType: []cryptosuite.SignatureType{"sigtype"}},
 		}
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
 		assert.Contains(t, err.Error(), "credential application's format must be a subset of the format property in the credential manifest")
 
 		// reset
 		ca.Format = &exchange.ClaimFormat{
 			JWT: &exchange.JWTType{Alg: []crypto.SignatureAlgorithm{crypto.EdDSA}},
 		}
 
-		def := exchange.PresentationDefinition{
-			ID: "test-id",
-			InputDescriptors: []exchange.InputDescriptor{
-				{
-					ID: "id-1",
-					Constraints: &exchange.Constraints{
-						Fields: []exchange.Field{
-							{
-								Path:    []string{"$.vc.issuer", "$.issuer"},
-								ID:      "issuer-input-descriptor",
-								Purpose: "need to check the issuer",
-							},
-						},
-					},
-				},
-			},
-		}
-
-		cm.PresentationDefinition = &def
 		ca.PresentationSubmission.DefinitionID = "badid"
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
-		assert.Contains(t, err.Error(), "credential application's presentation submission's definition id: test-id does not match the credential manifest's id: badid")
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
+		assert.Contains(t, err.Error(), "credential application's presentation submission's definition id: 32f54163-7166-48f1-93d8-ff217bdb0653 does not match the credential manifest's id: badid")
 
 		// reset
-		ca.PresentationSubmission.DefinitionID = def.ID
+		cm, ca, vcs = getValidTestCmCaVc(tt)
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
 		assert.NoError(tt, err)
 
 		ca.PresentationSubmission.DescriptorMap[0].Format = "badformat"
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
-		assert.Contains(t, err.Error(), "must be one of the following:")
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
+		assert.Contains(t, err.Error(), "format must be one of the following:")
 
 		// reset
 		ca.PresentationSubmission.DescriptorMap[0].Format = "jwt_vc"
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
 		assert.NoError(tt, err)
 
 		ca.PresentationSubmission.DescriptorMap[0].Path = "bad-path"
 
-		err = IsValidCredentialApplicationForManifest(cm, ca)
+		err = IsValidCredentialApplicationForManifest(cm, ca, vcs)
 		assert.Contains(t, err.Error(), "invalid json path: bad-path")
 
 	})
+
+	t.Run("Credential Application and Credential Manifest Pair VC id test", func(tt *testing.T) {
```

Comment:
it might be good to add a few specific types of input descriptors to check for - e.g. a test with a mismatched format, a test with a mismatched path, one where only some of the input descriptors were fulfilled, cases where multiple creds can fulfill multiple input descriptors, another where only one cred can fulfill multiple input descriptors...

Reply:
Added more tests